### PR TITLE
CIS: Domain Settings development

### PR DESCRIPTION
**Pull request by KumarGanesanIBM**

### CODE REVIEW

File: ibm/resource_ibm_cis_domain_settings.go [hunk 1]
```diff
@@ -2,268 +2,1075 @@ package ibm
...
+	"github.com/IBM/networking-go-sdk/zonessettingsv1"
 	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
 
-	v1 "github.com/IBM-Cloud/bluemix-go/api/cis/cisv1"
+const (
+	cisDomainSettingsDNSSEC                          = "dnssec"
+	cisDomainSettingsWAF                             = "waf"
+	cisDomainSettingsSSL                             = "ssl"
+	cisDomainSettingsUniversalSSL                    = "ssl_universal"
```
Comment (kavya498) [inline ibm/resource_ibm_cis_domain_settings.go:15]: `cisDomainSettingsUniversalSSL` is this attribute been supported in the resource.. as per our discussion we planned for new resource for tls settings...

Comment (KumarGanesanIBM) [reply ibm/resource_ibm_cis_domain_settings.go]: I forgot to remove this const. Removed it now.

File: ibm/resource_ibm_cis_domain_settings_test.go [hunk 2]
```diff
@@ -48,7 +75,7 @@ func testAccCheckCisSettingsConfigBasic3(id string, cisDomainStatic string) stri
 	return testAccCheckIBMCisDomainDataSourceConfigBasic1() + fmt.Sprintf(`
 	resource "ibm_cis_domain_settings" "%[1]s" {
 		cis_id          = data.ibm_cis.cis.id
-		domain_id       = data.ibm_cis_domain.cis_domain.id
+		domain_id       = data.ibm_cis_domain.cis_domain.domain_id
```
[the remark this answers is not in the document]
Comment (KumarGanesanIBM) [reply ibm/resource_ibm_cis_domain_settings_test.go:78]: it works both ways

File: ibm/resource_ibm_cis_domain_settings.go [hunk 3]
```diff
@@ -2,268 +2,1074 @@ package ibm
...
+	cisDomainSettingsWebSockets,
+	cisDomainSettingsChallengeTTL,
+	cisDomainSettingsMinify,
+	cisDomainSettingsSecurityHeader,
+	cisDomainSettingsMobileRedirect,
+	cisDomainSettingsMaxUpload,
+	cisDomainSettingsCipher,
+}
+
+func resourceCISSettingsCreate(d *schema.ResourceData, meta interface{}) error {
```
[the remark this answers is not in the document]
Comment (KumarGanesanIBM) [reply ibm/resource_ibm_cis_domain_settings.go]: done

**Pull request merged**